### PR TITLE
media-libs/libcanberra: Split out libcanberra-gtk3, depend on new virtual/sound-theme, prepare revdeps

### PR DESCRIPTION
**Pull request by a17r**
Maintainers, you are getting a notification for your package being prepared for a change in one of its dependencies.

TODO:
- [x] Split ACK
- [x] change all `media-libs/libcanberra[gtk3]` revdeps to `|| ( media-libs/libcanberra-gtk3 media-libs/libcanberra[gtk3(-)] )`
- [x] new package `virtual/sound-theme` for `media-libs/libcanberra` to depend on, accepting either `x11-themes/sound-theme-freedesktop` or `kde-plasma/ocean-sound-theme`

media-libs/libcanberra changes:
```diff
--- libcanberra-0.30-r7.ebuild
+++ libcanberra-0.30-r8.ebuild
@@ -1,7 +1,8 @@
 # Copyright 1999-2024 Gentoo Authors
 # Distributed under the terms of the GNU General Public License v2
 
-EAPI=7
+EAPI=8
+
 inherit libtool multilib-minimal systemd
 
 DESCRIPTION="Portable sound event library"
@@ -10,8 +11,8 @@
 
 LICENSE="LGPL-2.1+"
 SLOT="0"
-KEYWORDS="~alpha amd64 arm arm64 ~hppa ~loong ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux"
-IUSE="alsa gnome gstreamer gtk2 +gtk3 oss pulseaudio +sound tdb udev"
+KEYWORDS="~alpha ~amd64 ~arm ~arm64 ~hppa ~loong ~mips ~ppc ~ppc64 ~riscv ~s390 ~sparc ~x86 ~amd64-linux ~x86-linux"
+IUSE="alsa gstreamer oss pulseaudio tdb udev"
 
 DEPEND="
        dev-libs/libltdl:0[${MULTILIB_USEDEP}]
@@ -20,22 +21,11 @@
                media-libs/alsa-lib:=[${MULTILIB_USEDEP}]
                udev? ( virtual/libudev:=[${MULTILIB_USEDEP}] ) )
        gstreamer? ( media-libs/gstreamer:1.0[${MULTILIB_USEDEP}] )
-       gtk2? (
-               >=dev-libs/glib-2.32:2[${MULTILIB_USEDEP}]
-               >=x11-libs/gtk+-2.20.0:2[${MULTILIB_USEDEP}]
-               x11-libs/libX11[${MULTILIB_USEDEP}] )
-       gtk3? (
-               >=dev-libs/glib-2.32:2[${MULTILIB_USEDEP}]
-               x11-libs/gtk+:3[X,${MULTILIB_USEDEP}]
-               x11-libs/libX11[${MULTILIB_USEDEP}] )
        pulseaudio? ( media-libs/libpulse[${MULTILIB_USEDEP}] )
        tdb? ( sys-libs/tdb:=[${MULTILIB_USEDEP}] )
 "
 RDEPEND="${DEPEND}
-       gnome? (
-               gnome-base/dconf
-               gnome-base/gsettings-desktop-schemas )
-       sound? ( x11-themes/sound-theme-freedesktop )" # Required for index.theme wrt #323379
+       virtual/sound-theme" # Required for index.theme wrt #323379
 BDEPEND="
        app-arch/xz-utils
        virtual/pkgconfig
@@ -43,30 +33,28 @@
 
 REQUIRED_USE="udev? ( alsa )"
 
-PATCHES=(
-       # gtk: Don't assume all GdkDisplays are GdkX11Displays: broadway/wayland (from 'master')
-       "${FILESDIR}/${PN}-0.30-wayland.patch"
-)
-
 src_prepare() {
        default
        elibtoolize
 }
 
 multilib_src_configure() {
-       ECONF_SOURCE="${S}" econf \
-               --docdir="${EPREFIX}"/usr/share/doc/${PF} \
-               $(use_enable alsa) \
-               $(use_enable oss) \
-               $(use_enable pulseaudio pulse) \
-               $(use_enable gstreamer) \
-               $(use_enable gtk2 gtk) \
-               $(use_enable gtk3) \
-               $(use_enable tdb) \
-               $(use_enable udev) \
-               --disable-lynx \
-               --disable-gtk-doc \
+       local myeconfargs=(
+               --docdir="${EPREFIX}"/usr/share/doc/${PF}
+               $(use_enable alsa)
+               $(use_enable oss)
+               $(use_enable pulseaudio pulse)
+               $(use_enable gstreamer)
+               --disable-gtk
+               --disable-gtk3
+               $(use_enable tdb)
+               $(use_enable udev)
+               --disable-lynx
+               --disable-gtk-doc
                --with-systemdsystemunitdir="$(systemd_get_systemunitdir)"
+       )
+
+       ECONF_SOURCE="${S}" econf "${myeconfargs[@]}"
 
        if multilib_is_native_abi; then
                ln -s "${S}"/gtkdoc/html gtkdoc/html || die
@@ -81,8 +69,4 @@
 multilib_src_install_all() {
        einstalldocs
        find "${ED}" -type f -name '*.la' -delete || die
-
-       # This is needed for desktops different than GNOME, bug #520550
-       exeinto /etc/X11/xinit/xinitrc.d
-       newexe "${FILESDIR}"/libcanberra-gtk-module.sh 40-libcanberra-gtk-module
 }
```

### CODE REVIEW

**Review by xgqt — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

**Review by fordfrog — APPROVED** [review]

File: virtual/sound-theme/sound-theme-0.ebuild [hunk 1]
```diff
@@ -0,0 +1,18 @@
+# Copyright 1999-2024 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=8
+
+DESCRIPTION="Virtual for themes following the freedesktop.org sound naming spec"
+
+SLOT="0"
+KEYWORDS="~alpha amd64 arm arm64 ~hppa ~loong ~mips ppc ppc64 ~riscv ~s390 sparc x86 ~amd64-linux ~x86-linux"
+IUSE="+sound"
```
Comment (thesamesam) [inline virtual/sound-theme/sound-theme-0.ebuild:10]: Why does it have a USE at all?

Also, new virtuals have to go to the ML.

Comment (a17r) [reply virtual/sound-theme/sound-theme-0.ebuild:10]: Because it was RDEPEND optional in libcanberra before, and this was just moved from there.

Comment (a17r) [reply virtual/sound-theme/sound-theme-0.ebuild:10]: See also: https://bugs.gentoo.org/941986#c2

@pacho2 ^

**Review by Flowdalic — APPROVED** [review]

**Review by thesamesam — APPROVED** [review]

**Comment by a17r** [comment]
I will likely need to drop the new virtual and treat the freedesktop theme more like hicolor icon theme instead. It is supposed to be an always present fallback per the spec, and apparently applications can install custom sounds in there (no idea of an example). That's what I got from Plasma upstream devs.

**Pull request merged**